### DISCUSSION TITLE
Welcome to spring-blog Discussions!

### DISCUSSION THREAD

**Discussion by sivaprasadreddy** [sivaprasadreddy/spring-blog#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by BHIMAVARAPU-MANOJ-KUMAR** [comment]
**🎉 Hello, Community! 🎉**

I’m **Manoj Kumar Reddy**. I'm from **Vijayawada, Andhra Pradesh, India**. I’m excited to be a part of this Community! 

I’ve been working with **Java technology**, **Spring Framework**, **Spring Boot Microservices**, **Kafka**, **RabbitMQ** and tools like **Postman**, **Docker** etc., and am looking forward to contribute to this **spring-blog community project**.

**About me:** I have a background in **Java** based **Backend development**, and some of my recent projects include **[Event-Management-System](https://github.com/BHIMAVARAPU-MANOJ-KUMAR/Event-Management-System-GyanGrove-Backend-Intern-Assignment)**, other topics like **Kafka, RabbitMQ AMQP** and I work regularly with **Spring Framework Spring Boot**.

**Why I'm Here:** I’m passionate about **building robust APIs**, **community engagement** and am committed to helping this project grow. I’m also hoping to connect with and learn from other contributors here! 😊

Please feel free to reach out if you have ideas, or just want to connect!

Looking forward to collaborating with all of you!

**[BHIMAVARAPU MANOJ KUMAR REDDY](https://github.com/BHIMAVARAPU-MANOJ-KUMAR)**

**Comment by sivaprasadreddy** [comment]
Hi @BHIMAVARAPU-MANOJ-KUMAR 
Thanks for joining and showing the interest to contribute to the project.

Please take a look at the Issues list and add a comment if you want to work on any issue.

**Comment by BHIMAVARAPU-MANOJ-KUMAR** [comment]
**A dedicated Frontend Application using Angular or React ?**

**Why to not implement a dedicated Frontend Application using Angular or React ? @sivaprasadreddy Any particular Reason ?**

**Reply by sivaprasadreddy** [reply]
To me using Thymeleaf with HTMX is sufficient for the application and less management(no need to have 2 application, npm build setup, etc).

Is there any benefit of separating the frontend for this application?

**Reply by BHIMAVARAPU-MANOJ-KUMAR** [reply]
**Okay @sivaprasadreddy. I've to look into what HTMX is..!**

**Comment by BHIMAVARAPU-MANOJ-KUMAR** [comment]
Hey **@sivaprasadreddy**, today I've cloned **Spring Blog Project** on to my Local Development System. Went through the code, came to know that the project is a **`Package by Feature`**. Why can't we use **[Project Lombok](https://projectlombok.org/)**, to reduce some lines of code like **`getters and setters etc.,`** especially in the Entity Classes.

**Reply by sivaprasadreddy** [reply]
I find Lombok useful to reduce boilerplate code. But, it poses some problems when used in open-source projects:

* Different people may use different IDEs and in some IDEs installing Lombok plugin is a hassle
* Lombok alone works fine, but if we use other annotation processors (ex: https://errorprone.info/) then there are some issues with Lombok
* It's easy to misuse Lombok while working with JPA, especially for newbies.

Due to these reasons I prefer not to use Lombok.

**Reply by BHIMAVARAPU-MANOJ-KUMAR** [reply]
Okay got your point **@sivaprasadreddy**
